lib/logstash/inputs/heartbeat.rb

```diff
@@ -38,34 +41,74 @@ class LogStash::Inputs::Heartbeat < LogStash::Inputs::Threadable
   # This is typically used only for testing purposes.
   config :count, :validate => :number, :default => -1
 
+  # Select the type of sequence, deprecating the 'epoch' and 'sequence' values in 'message'.
+  #
+  # If you set this to `epoch` then this plugin will use the current
+  # timestamp in unix timestamp (which is by definition, UTC).
+  #
+  # If you set this to `sequence` then this plugin will send a sequence of
+  # numbers beginning at 0 and incrementing each interval.
+  #
+  # If you set this to 'none' then no field is created
+  config :sequence, :validate => ["none", "epoch", "sequence"]
+
   def register
     @host = Socket.gethostname
+    @field_sequence = ecs_select[disabled: "clock", v1: "[event][sequence]"]
+    if sequence.nil? && ["epoch", "sequence"].include?(message)
+      logger.warn("message contains sequence type specification (epoch|sequence) for this purpose use the sequence option")
+    end
+    if ecs_compatibility == :disabled
+      unless sequence.nil?
+        @sequence_selector = decode_sequence_selector(sequence)
+        logger.debug("Using sequence as sequence selector")
+      else
+        @sequence_selector = decode_sequence_selector(message)
+        logger.debug("Using message as sequence selector")
+      end
+    else
+      @sequence_selector = decode_sequence_selector(sequence)
+      if ["epoch", "sequence"].include?(message)
+        logger.warn("message setting still contains 'epoch' or 'sequence' selectors which is not considered for for configuring the plugin")
+      end
+    end
+    @valid_message_payload = message && "epoch" !=  message && "sequence" != message
   end
 
   def run(queue)
-    sequence = 0
+    sequence_count = 0
 
     while !stop?
       start = Time.now
 
-      sequence += 1
-      event = generate_message(sequence)
+      sequence_count += 1
+      event = generate_message(sequence_count)
       decorate(event)
       queue << event
-      break if sequence == @count || stop?
+      break if sequence_count == @count || stop?
 
       sleep_for = @interval - (Time.now - start)
       Stud.stoppable_sleep(sleep_for) { stop? } if sleep_for > 0
     end
   end
 
-  def generate_message(sequence)
-    if @message == "epoch"
-      LogStash::Event.new("clock" => Time.now.to_i, "host" => @host)
-    elsif @message == "sequence"
-      LogStash::Event.new("clock" => sequence, "host" => @host)
-    else
-      LogStash::Event.new("message" => @message, "host" => @host)
+  def generate_message(sequence_count)
+    if @sequence_selector == :none
+      return LogStash::Event.new("message" => @message, "host" => @host)
     end
+
+    sequence_value = @sequence_selector == :epoch ? Time.now.to_i : sequence_count
+    evt = LogStash::Event.new("host" => @host)
+    evt.set(@field_sequence, sequence_value)
+    evt.set("message", @message) if @valid_message_payload
```

Comment:
If you take my other suggestion about intercepting magic messages, this becomes:

```suggestion
    evt.set("message", @message) unless @message.nil?
```